src/acquisition/covidcast/csv_to_database.py

```diff
@@ -0,0 +1,127 @@
+"""Imports covidcast CSVs and stores them in the epidata database."""
+
+# standard library
+import argparse
+import os
+
+# first party
+from delphi.epidata.acquisition.covidcast.csv_importer import CsvImporter
+from delphi.epidata.acquisition.covidcast.database import Database
+from delphi.epidata.acquisition.covidcast.file_archiver import FileArchiver
+
+
+def get_argument_parser():
+  """Define command line arguments."""
+
+  parser = argparse.ArgumentParser()
+  parser.add_argument(
+    '--data_dir',
+    help='top-level directory where CSVs are stored')
+  parser.add_argument(
+    '--test',
+    action='store_true',
+    help='do a dry run without committing changes')
+  return parser
+
+
+def scan_upload_archive(
+    data_dir,
+    database,
+    csv_importer_impl=CsvImporter,
+    file_archiver_impl=FileArchiver):
+  """Find CSVs, upload them to the database, and archive them.
+
+  data_dir: top-level directory where CSVs are stored
+  database: an open connection to the epidata database
+
+  The CSV storage layout is assumed to be as follows:
+
+  - Receiving: <data_dir>/receiving/<source name>/*.csv
+  - Archival: <data_dir>/archive/<status>/<source name>/*.csv[.gz]
+
+  Status above is one of `successful` or `failed`. See the accompanying readme
+  for further details.
+  """
+
+  receiving_dir = os.path.join(data_dir, 'receiving')
+  archive_successful_dir = os.path.join(data_dir, 'archive', 'successful')
+  archive_failed_dir = os.path.join(data_dir, 'archive', 'failed')
+
+  # helper to archive a failed file without compression
+  def archive_as_failed(path_src, filename, source):
+    print('archiving as failed')
+    path_dst = os.path.join(archive_failed_dir, source)
+    compress = False
+    file_archiver_impl.archive_file(path_src, path_dst, filename, compress)
+
+  # helper to archive a successful file with compression
+  def archive_as_successful(path_src, filename, source):
+    print('archiving as successful')
+    path_dst = os.path.join(archive_successful_dir, source)
+    compress = True
+    file_archiver_impl.archive_file(path_src, path_dst, filename, compress)
```

Comment:
i do worry a little bit more about the potential silent overwrite in this scenario. for example, if you somehow get a truncated version of a file that was already ingested, having both versions archived could be useful for both determining the extent of the problem and for quickly back-filling.

Reply:
yeah, completely agree. could i ask a favor of filing an issue for this? probably worth fixing in the long-run, but delphi needs something functional asap.

Reply:
will do.

Reply:
https://github.com/cmu-delphi/delphi-epidata/issues/73